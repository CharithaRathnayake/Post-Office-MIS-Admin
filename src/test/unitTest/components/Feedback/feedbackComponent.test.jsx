import React from "react";
import  ReactDOM  from "react-dom";
import FeedbackComponent from "../../../../components/Complain/complainComponent";
import {render,cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect"
import renderer from "react-test-renderer";

describe("Feedback Form",()=>{
    afterEach(cleanup);

    it ("renders without crashing",()=>{
        const div = document.createElement("div");
        ReactDOM.render(<FeedbackComponent/>,div)
    })

    it ("renders title correctly",()=>{
        const {getByTestId}= render(<FeedbackComponent/>);
        expect(getByTestId("title-id")).toHaveTextContent("Title:-")
    })

    it ("matches snapshot",()=>{
        const tree = renderer.create(<FeedbackComponent/>).toJSON();
        expect(tree).toMatchSnapshot();
        
    })
})