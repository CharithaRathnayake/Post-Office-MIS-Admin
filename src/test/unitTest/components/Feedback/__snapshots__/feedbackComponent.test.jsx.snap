// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback Form matches snapshot 1`] = `
<div
  className="makeStyles-form-1"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div
        className="makeStyles-title-2"
        data-testid="title-id"
      >
        Title:-
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div>
        PID: 
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div>
        Name: 
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div>
        Date: 
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div>
        Content:  ----ahcbakbk nkjcnkasjnkas
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div
        className="makeStyles-button-3"
      >
        <button>
          Mark as Resloved
        </button>
      </div>
    </div>
  </div>
</div>
`;
